style.md

```diff
@@ -1923,6 +1924,140 @@ sptr := &T{Name: "bar"}
 </td></tr>
 </tbody></table>
 
+### Initializing Maps
+
+Prefer `make(..)` for empty maps, and maps populated
+programmatically. This makes map initialization visually
+distinct from declaration, and it makes it easy to add size
+hints later if available.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+var (
+  // m1 is safe to read and write;
+  // m2 will panic on writes.
+  m1 = map[T1]T2{}
+  m2 map[T1]T2
+)
+```
+
+</td><td>
+
+```go
+var (
+  // m1 is safe to read and write;
+  // m2 will panic on writes.
+  m1 = make(map[T1]T2)
+  m2 map[T1]T2
+)
+```
+
+</td></tr>
+<tr><td>
+
+Declaration and initialization are visually similar.
+
+</td><td>
+
+Declaration and initialization are visually distinct.
+
+</td></tr>
+</tbody></table>
+
+Where possible, provide capacity hints when initializing
+maps with `make()`.
+
+```go
+make(map[T1]T2, hint)
+```
+
+While Go's runtime map implementation only acts upon capacity
+hints in certain situations (see
+[`runtime.makemap()`](https://golang.org/src/runtime/map.go)
+for more information), this can also be a helpful cue when
+reading source code.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+m := make(map[T1]T2)
+
+// later...
+
+for i := 0; i < 10000; i++ {
+  // add element to m
+)
+```
+
+</td><td>
+
+```go
+m := make(map[T1]T2, 10000)
+
+// later...
+
+for i := 0; i < 10000; i++ {
+  // add element to m
+}
+```
+
+</td></tr>
+<tr><td>
+
+The footprint of `m` is not well-understood at initialization time,
+and multiple allocations may take place as part of adding elements.
+
+</td><td>
+
+The footprint of `m` is better understood at initialization time,
+and fewer allocations are incurred when adding within its capacity.
+
+</td></tr>
+</tbody></table>
+
+On the other hand, if the map holds a fixed list of elements,
+use map literals to initialize the map.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+m := make(map[T1]T2, 3)
+m[k1] = v1
+m[k2] = v2
+m[k3] = v3
+```
+
+</td><td>
+
+```go
+m := map[T1]T2{
+  k1: v1,
+  k2: v2,
+  k3: v3,
+}
+```
+
+</td></tr>
+</tbody></table>
+
+
+The basic rule of thumb is to use map literals when adding a fixed set of
+elements at initialization time, otherwise use `make` (and specify a size hint
+if available).
+
+There are more nuances than outlined in this guide; when in doubt,
```

Comment:
optional: OK to have this, but think it could be dropped without much of an impact, since we've called out "basic rule of thumb" above.

Reply:
Probably right. Will remove.